packages/hdwallet-provider/src/index.ts

```diff
@@ -279,30 +279,28 @@ class HDWalletProvider {
           // always resolve the Promise - if an error occurs, throw it when
           // this.initialized is checked so errors can be handled by the user
           if (error) {
-            resolve({
+            return resolve({
               success: false,
               error
             });
-            return;
           } else if (response.error) {
```

Comment:
You can also get rid of the else here ... and use `if (response.error) ...`, your call, though.

Reply:
I actually don't really like that style-wise but thanks for the suggestion!